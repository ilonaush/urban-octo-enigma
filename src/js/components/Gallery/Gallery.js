import React from 'react';
import {connect} from "react-redux";
import LazyLoadImage from "../../tools/LazyLoad"
import "./Gallery.styl"
import PropTypes from 'prop-types';


export const Gallery = ({cats}) =>  {
    return (
        <div className='workers-gallery'>
            {cats && cats.length ?
                cats.map((cat) => {
                    return (
                    <div className='worker-photo' key={cat.id}>
                        <LazyLoadImage height={200} src={`cat-avatars/${cat.img}`}/>
                        <div className='worker-name'>{cat.name}</div>
                        <div className='worker-position'>{cat.age}</div>
                    </div>
                    )
                })
                : <div>Немає котиків</div>
            }
        </div>
    );
};

Gallery.propTypes = {
        cats: PropTypes.array
};

Gallery.defaultProps = {
    cats: []
};

export default connect((state) => ({cats: state.cats}))(Gallery);


