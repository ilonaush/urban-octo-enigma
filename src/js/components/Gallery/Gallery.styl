.workers-gallery
  display flex
  flex-wrap wrap
  .worker-photo
    width 350px
    margin 10px