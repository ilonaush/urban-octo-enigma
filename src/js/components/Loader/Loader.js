import React from 'react';
import "./Loader.styl"

const Loader = () =>
    <div className='loader'>
        Loading...
    </div>
;

export default Loader;
