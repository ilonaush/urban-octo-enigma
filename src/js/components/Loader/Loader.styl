@import "~js/vars";

.loader
  background-color rgba(255,255,255, .5)
  position absolute
  top 0
  left 0
  display flex
  align-items center
  font-size 50px
  text-align center
  width 100%
  justify-content center
  height 100%
  color $orange
