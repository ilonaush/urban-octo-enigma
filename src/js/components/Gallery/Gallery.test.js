import React from 'react';
import Gallery  from './Gallery';
import ReactDOM from 'react-dom';
import {Provider} from "react-redux";
import renderer from 'react-test-renderer';
import Loader from "../Loader/Loader";
import configureMockStore from 'redux-mock-store'
import { shallow, mount, render } from 'enzyme';

const mockStore = configureMockStore();
const store = mockStore({ workers: [{id: 1, fullname: 'mike', position: 'cat'}] });
const emptyStore = mockStore({});


it('renders correctly gallery component with mock store', () => {
    const GalleryComponent = renderer.create(<Provider store={store}><Gallery/></Provider>).toJSON();
    expect(GalleryComponent).toMatchSnapshot();
});


it('renders correctly gallery component with default parameters', () => {
    const GalleryComponent = mount(<Provider store={emptyStore}><Gallery/></Provider>);
    expect(Array.isArray([(GalleryComponent).prop('workers')])).toBe(true);
});