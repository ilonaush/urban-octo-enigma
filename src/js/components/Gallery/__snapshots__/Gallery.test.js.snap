// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly gallery component 1`] = `
<div
  className="workers-gallery"
>
  <div
    className="worker-photo"
  >
    <div>
      mike
    </div>
  </div>
</div>
`;
